crates/bevy_sprite/src/frustum_culling.rs

```diff
@@ -0,0 +1,126 @@
+use bevy_asset::{Assets, Handle};
+use bevy_ecs::{Commands, Entity, Query, Res, With};
+use bevy_math::Vec2;
+use bevy_render::{camera::Camera, draw::Frustum};
+use bevy_transform::components::Transform;
+use bevy_window::Windows;
+
+use crate::{Sprite, TextureAtlas, TextureAtlasSprite};
+
+struct Rect {
+    position: Vec2,
+    size: Vec2,
+}
+
+impl Rect {
+    #[inline]
+    pub fn is_overlapping(&self, other: Rect) -> bool {
+        !(self.position.x > other.end().x
+            || self.end().x < other.position.x
+            || self.position.y > other.end().y
+            || self.end().y < other.position.y)
+    }
+
+    #[inline]
+    pub fn end(&self) -> Vec2 {
+        self.position + self.size
+    }
+}
+pub fn frustum_culling_sprites(
+    commands: &mut Commands,
+    windows: Res<Windows>,
+    cameras: Query<&Transform, With<Camera>>,
+    visible: Query<&Frustum, With<Sprite>>,
+    sprites: Query<(Entity, &Transform, &Sprite)>,
+) {
+    let window = windows.get_primary().unwrap();
+    let window_size = Vec2::new(window.width(), window.height());
+
+    for camera_transform in cameras.iter() {
+        let camera_position = Vec2::new(
+            camera_transform.translation.x,
+            camera_transform.translation.y,
+        );
+
+        let camera_size = window_size * camera_transform.scale.truncate();
+
+        let rect = Rect {
+            position: camera_position - camera_size / Vec2::new(2.0, 2.0),
+            size: camera_size,
+        };
+
+        for (entity, drawable_transform, sprite) in sprites.iter() {
```

Comment:
I think you can convert this into a parallel iterator, which might give a boost if you expect to iterate over a large number of items.
```rust
.par_iter_mut(32).for_each(
    &pool,
    |(entity, drawable_transform, sprite)| {
        // Do stuff
);
```